File: Backend/src/ShoeStore.Application/DTOs/PageResult.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ShoeStore.Application.DTOs
{
    public class PageResult<T>
    {
        public IEnumerable<T> Items { get; set; } = new List<T>();

        public int TotalCount { get; set; }

        public int PageNumber { get; set; }

        public int PageSize { get; set; } = 10;

        public int TotalPages => PageSize > 0 ? (int)Math.Ceiling(TotalCount / (double)PageSize) : 0;

        public bool HasPrevious => PageNumber > 1;
        public bool HasNext => PageNumber < TotalPages;
    }
}

File: Frontend/app/src/main/java/com/example/shoestoreapp/features/agent_intelligent/data/remote/PageResult.kt
package com.example.shoestoreapp.features.agent_intelligent.data.remote

import com.google.gson.annotations.SerializedName

data class PageResult<T>(
    @SerializedName("items")
    val items: List<T>,
    @SerializedName("totalCount")
    val totalCount: Int,
    @SerializedName("pageNumber")
    val pageNumber: Int,
    @SerializedName("pageSize")
    val pageSize: Int,
    @SerializedName("totalPages")
    val totalPages: Int,
    @SerializedName("hasPrevious")
    val hasPrevious: Boolean,
    @SerializedName("hasNext")
    val hasNext: Boolean
) {
}